### PR TITLE
Suppress warnings and add test coverage

### PR DESCRIPTION
**Pull request by shoyer**
- Suppressed various warnings.
- PsuedoNetCDF was not getting tested. Now it is.
- Added test coverage for `print_versions.py`

### CODE REVIEW

File: ci/requirements/py36.yml [hunk 1]
```diff
@@ -8,7 +8,7 @@ dependencies:
   - distributed
   - h5py
   - h5netcdf
-  - matplotlib
+  - matplotlib=3.0  # for https://github.com/barronh/pseudonetcdf/issues/69
```
Comment (crusaderky) [inline ci/requirements/py36.yml]: Why only py36.yml? What about the other requirements files?
Also, will this restriction appear in the conda recipe?

Comment (shoyer) [reply ci/requirements/py36.yml]: This is just to ensure that we have at least one test job testing xarray with psuedonetcdf, which doesn't import in matplotlib 3.1 due to the linked issue. Xarray itself works perfectly fine on matplotlib 3.1.

**Comment by dcherian** [comment]
I'm also seeing
```
xarray/core/dataset.py:3666
  /home/deepak/work/python/xarray/xarray/core/dataset.py:3666: FutureWarning: Converting timezone-aware DatetimeArray to timezone-naive ndarray with 'datetime64[ns]' dtype. In the future, this will return an ndarray with 'object' dtype where each element is a 'pandas.Timestamp' with the correct 'tz'.
  	To accept the future behavior, pass 'dtype=object'.
  	To keep the old behavior, pass 'dtype="datetime64[ns]"'.
    data = np.asarray(series).reshape(shape)

/home/deepak/miniconda3/envs/dcpy/lib/python3.7/site-packages/pandas/core/apply.py:286
  /home/deepak/miniconda3/envs/dcpy/lib/python3.7/site-packages/pandas/core/apply.py:286: FutureWarning: Converting timezone-aware DatetimeArray to timezone-naive ndarray with 'datetime64[ns]' dtype. In the future, this will return an ndarray with 'object' dtype where each element is a 'pandas.Timestamp' with the correct 'tz'.
  	To accept the future behavior, pass 'dtype=object'.
  	To keep the old behavior, pass 'dtype="datetime64[ns]"'.
    results[i] = self.f(v)
```

locally.

**Comment by shoyer** [comment]
Yeah, I'm not quite sure how we want to handle the datetime64 with timezones warnings. Possibly we just want to silence them in our test suite, and enocourage users to explicitly cast their pandas data to a dtype supported by numpy before loading it into xarray. Doing that properly would require the equivalent of "re-raising" a warning.

**Comment by shoyer** [comment]
I'm going to merge this shortly, and leave further clean-up for later.

**Pull request merged**